### PR TITLE
admin: Display repo sync counters per sync job

### PR DESCRIPTION
**Pull request by ryanslade**
Expose sync counters in the UI per external service sync job.

Here's a short demo:

https://user-images.githubusercontent.com/25610/196177127-476de002-5786-4fd3-a89a-4a52c469c4db.mov

Closes https://github.com/sourcegraph/sourcegraph/issues/42868

Thanks @sashaostrikov for the help figuring out how to get the `ValueLegendList` component integrated.

## Test plan

Tested locally, see video above

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (+0.13 kb) | 0.03% (+4.16 kb) | 0.03% (+4.04 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/rs/display-sync-counters/compare-report.html;tab=live_object) for a full comparison between the commits 828e34d and b3cbf2e or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by mrnugget — APPROVED** [review]

**Review by eseliger — APPROVED** [review]
wow this is so cool!

**Review by sashaostrikov — APPROVED** [review]
Let's gooooooo! :shipit::shipit::shipit:

(I'd engage designers just in case, but to my preference this looks awesome!)

File: client/web/src/components/externalServices/backend.ts [hunk 1]
```diff
@@ -312,6 +312,12 @@ export const EXTERNAL_SERVICE_SYNC_JOBS = gql`
         startedAt
         finishedAt
         failureMessage
+        reposSynced
+        repoSyncErrors
+        reposAdded
+        reposDeleted
+        reposModified
+        reposUnmodified
```
Comment (pjlast) [inline client/web/src/components/externalServices/backend.ts:320]: You need to update `client/web/src/components/externalServices/ExternalServicePage.story.tsx` , line 54, to contain these fields as well

**Pull request merged**